src/snowflake/snowpark/udaf.py

```diff
@@ -110,7 +113,11 @@ def __call__(
 
         udaf_expr = None
         if _emit_ast and self._ast is not None:
-            assert self._ast_id is not None, "Need to assign UDAF an ID."
+            if self._ast_id is None:
+                session = (
+                    self._session or snowflake.snowpark.session._get_active_session()
+                )
```

Comment:
is it possible session is still None here?

Reply:
As far as I can tell, if you try to call a UDxF without an active session, it means it will not have been registered in the first place.

This is likely moving the eventual failure slightly earlier in the pipeline, where if the session is still None here, `_get_active_session()` will raise an error.